src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -197,7 +210,7 @@ private static String getCourseName(@NotNull JSONObject jsonObject, @NotNull Str
     for (int i = 0; i < modulesJsonArray.length(); ++i) {
       try {
         JSONObject moduleObject = modulesJsonArray.getJSONObject(i);
-        modules.add(Module.fromJsonObject(moduleObject));
+        modules.add(Module.fromJsonObject(moduleObject, factory));
       } catch (JSONException ex) {
```

Comment:
Nit: a tiny styling notice, sometimes we use `e` for `Exception` and sometimes `ex`, shall we unify (I suggest to `ex`, as we have `e` for `events` in other places)

Reply:
It's a good idea to unify this. I started with `ex` but then, at some point, it struck my mind that `ex` must be C# convention and in Java, `e` is used instead, and so I changed to use `e` but was too lazy to revise those I wrote earlier.

You are probably right in that either can be used and your argument favoring `ex` is good. But I suggest we leave this issue out of this PR as there is already so much stuff in it.

Reply:
this is the case where C# approach is fully applicable :)